src/transformers/modeling_tf_t5.py

```diff
@@ -355,16 +355,33 @@ def call(
         return outputs  # hidden-states, (self-attention weights), (self-attention position bias), (cross-attention weights), (cross-attention position bias)
 
 
+class _NoLayerEmbedTokens(object):
+    # this class wraps a the TFSharedEmbeddingTokens layer into a
```

Comment:
That's important! If the class `TFT5MainLayer` saved `self.shared` directly without a "NoLayerWrapper" it expects weights to be loaded into this layer (for example when doing `.from_pretrained(...)`), but we just want to reuse the weights as saved in the main class TFT5Model without creating "new" weights @craffel @thomwolf . Thanks to @mfuntowicz for the abstraction idea

Reply:
Idea behind this wrapper is:

- Every subclass of `Layer` register in `self.<some_variable>` is automatically registered as a learnable component which should expose `weight` attribute. 

- Wrapping a Layer into an object allows this underlying layer to not "belong" to the set of parameters on `self` and thus doesn't attempt to restore store weight on it.